blog/2018-01-09-smart-testing-for-fast-feedback.textile

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: Smart Testing For Fast Feedback
+author: hemanik
+tags: [ testing, guide, smart-testing ]
+---
+
+h1. Smart Testing For Fast Feedback
+
+p<>. ??As an engineer, you should constantly work to make your feedback loops shorter in time and/or wider in scope.??
+
+p>. ??"-@KentBeck":https://twitter.com/KentBeck/status/531964254946328576??
+
+
+One of the biggest issues that plague the software development life cycle is the slow feedback loop, the time between changing code (or tests) and getting results from the tests.
+
+h2. Impact of Slow Feedback Loop
+
+Consider you are a team doing "Test Driven Development(TDD)":https://en.wikipedia.org/wiki/Test-driven_development, then the general workflow is you write some tests, add code to make those tests pass and run those tests to see if the tests pass and refactor if the tests fail and repeat the process.
+
+This vicious cycle of programming and testing doesn’t just slow you down in terms of time spent (especially with long-running tests), the involved "context switching":https://www.petrikainulainen.net/software-development/processes/the-cost-of-context-switching/ also drains your mental energy, which might ultimately destroy your productivity.
+
+Here, fast feedback means that the time between changing code (or tests) and getting results from running the tests is reduced to a minimum. In other words, you end up with a fast edit-compile-test loop. The great thing about this, as "Joel Spolsky put it":http://www.joelonsoftware.com/articles/fog0000000023.html: “the faster the Edit-Compile-Test loop, the more productive you will be”.
+
+For almost every new feature added to the application, there is also a test written. This way, whenever someone changes the user interface, the system will be able to catch the differences and signal the team about any issues. They're still not integrated into the main pipeline. The plan is to merge them in to gain more confidence in what they build. The tests should run whenever anyone pushes changes to the git repository.
+
+Work starts on the new CD pipeline and everything is set up accordingly. The pipeline structure looks like below:
```

Comment:
I'm missing the purpose of the next lines - it should be a description of the developer's workflow? I'm not sure if it is really necessary to describe the whole process here, just highlight the most important things...